typescript/cli/src/deploy/warp.ts

```diff
@@ -597,13 +649,27 @@ async function enrollRemoteRouters(
       );
       const mutatedConfigTxs: AnnotatedEV5Transaction[] =
         await evmERC20WarpModule.update(mutatedWarpRouteConfig);
-      for (const transaction of mutatedConfigTxs) {
-        const receipt: ContractReceipt = await multiProvider.sendTransaction(
-          chain,
-          transaction,
+
+      if (mutatedConfigTxs.length == 0)
+        return logGreen(
+          `Mutated warp config on ${chain} is the same as target. No updates needed.`,
+        );
+
+      const transactionReceipts = await submitter.submit(...mutatedConfigTxs);
+
+      if (transactionReceipts && transactionReceipts.length > 0) {
+        if (warpApplyParams.chainSubmissionStrategy)
+          return logGreen(
+            `✅ Router enrollment update successfully submitted to safe (${warpApplyParams.chainSubmissionStrategy}) on ${chain}:\n\n`,
+            indentYamlOrJson(yamlStringify(transactionReceipts, null, 2), 4),
+          );
```

Comment:
ditto
also can you help me understand why we have this code in multiple places?

Reply:
Removed

Reply:
Because we want to submit a set of tx for each chain. 

If we DRY this by putting all the txs into a single transactions array and then submitting it at the end, we'd have to group them by chain, then look up the strategy associated for that chain, and then send it.

Reply:
We also submit the safe txs as a batch[]. we could do them individually in a loop. although i like it DRY-er, i'll leave this for now